content/docs/user-guide/assets/runtime.md

```diff
@@ -0,0 +1,48 @@
+
+## Loading Assets at Runtime
+
+The [AZ::Data::AssetManager](https://github.com/o3de/o3de/blob/18205539abf1b1d2eb3959c0a1c42a3eea16a455/Code/Framework/AzCore/AzCore/Asset/AssetManager.h#L136) system is responsible for loading and managing assets at runtime.
+
+`GetAsset` is the primary API used to load an existing asset.  It returns an `Asset<T>` reference.  Only one copy of an asset is loaded in memory at a time, so if an asset is already loaded or currently loading, no new load requests will be made; the `Asset<T>` returned will point to the same `AssetData` and will be ready when the load is finished (if it hasn't already finished).  `GetAsset` is a non-blocking asychronous API, meaning the asset data may not be available immediately after the call completes.
+
+There are two ways to handle waiting for an asset load to complete: the blocking way is to call [BlockUntilLoadComplete](https://github.com/o3de/o3de/blob/18205539abf1b1d2eb3959c0a1c42a3eea16a455/Code/Framework/AzCore/AzCore/Asset/AssetCommon.h#L387) on the `Asset<T>` reference.  This will block the current thread until loading of the asset is completed.  The other (and recommended) option is to connect to the [AssetBus](https://github.com/o3de/o3de/blob/18205539abf1b1d2eb3959c0a1c42a3eea16a455/Code/Framework/AzCore/AzCore/Asset/AssetCommon.h#L527) and handle the `OnAssetReady`/`OnAssetReloaded` events for your asset.
+
+Note that the engine does not currently support load cancelling (cancelling a load request before completion), making it possible to call GetAsset and then discard the result while waiting for `OnAssetReady` to be called.  This is highly discouraged as it would stop functioning if/when load cancelling support is added.
+
+### Dependency Loading
+
+By default, the `AssetManager` will load all the dependencies of an asset before signalling an asset has completed loading.  Asset dependencies are recorded in the `AssetCatalog` and are determined by the Product Dependencies emitted by each builder.  There are 3 different load settings possible for dependencies (`AssetLoadBehavior`):
+1. Preload - The dependency will complete loading before the parent asset is signalled ready.
+2. QueueLoad - The dependency will start loading when the parent asset starts but will not be required to complete before the parent asset has been signalled ready.
+3. NoLoad - The dependency will not be requested to load.  If the dependency is already loaded anyway, any references will still point to the already-loaded asset.
+
+Keep in mind that regardless of the settings, AssetManager makes no guarantees on the order assets complete loading.
+
+### Hot-Reloading
+

```

Comment:
I also suggest adding a note on the most common gotcha for hot reload events:

Often, projects to copy data out of loaded assets to perform additional processing. A common error around asset reloading is to not refresh any copied data. For example, if your project is a game that uses an XML file to define the player's movement speed, and when this XML file is loaded, the information is copied to a component on the player entity, then when a reload event occurs, it's important to also re-copy the information over to the player entity. Otherwise, the player will continue to move at the old movement speed, even if the XML file is updated to change it.